wordpress-mobile/WordPress-iOS
# Use 'cocoapods-repo-update' CocoaPods plugin

I have put together a CocoaPods plugin called `cocoapods-repo-update` here: https://github.com/wordpress-mobile/cocoapods-repo-update.

The purpose of this is to check if the local specs repo needs to be updated when `pod install` is run. This is particularly important for us because BuddyBuild does not provide a way of updating the repo.

**Note:** Since this change relies on the `cocoapods-repo-update` gem, `pod install` now requires being run with `bundle exec`. I have added improving the behaviour of `rake dependencies` to my to-do list.

To test:

- Run `bundle install`
- Run `bundle exec pod install`
- You should see `Not updating local specs repo` in the output.

To test the specs update behaviour, find a newly released pod in https://github.com/CocoaPods/Specs/commits/master and add it to the `Podfile` (I used `pod 'IceCream', '1.7.2'`).

- Run `bundle exec pod install` again.
- You should see `At least one Pod is not in the local specs repo. Updating specs repo...` in the output and it should install the new pod successfully.

I have also tested this successfully using BuddyBuild in [this build](https://dashboard.buddybuild.com/apps/57a120bbe0f5520100e11c19/build/5beaf3ee2deb720001bc96d6#logs).

cc @stevebaranski
